### ISSUE TITLE
[BUG] 2.0.5 breaks with python 3.5.2 Linux

### ISSUE THREAD

**Issue by nbgooch** [Ousret/charset_normalizer#105]
**Describe the bug**

Charset normalizer 2.0.5 breaks with python 3.5.2 on `ubuntu:trusty`. We have some Jenkins job that have been running successfully for months and then on 9/14 everything broke.

**To Reproduce**

This is an enterprise application so I can't give you source, but we run
```
pip3 install hvac --user --disable-pip-version-check
python3 asoc.py
```
I believe you should be able to recreate with a sample app using hvac.

**Expected behavior**

For this not to happen :smile
```
python3 asoc.py
Traceback (most recent call last):
  File "/home/jenkins/.local/lib/python3.5/site-packages/charset_normalizer/api.py", line 5, in <module>
    from os import PathLike
ImportError: cannot import name 'PathLike'
```

**Logs**

```
+ pip3 install hvac --user --disable-pip-version-check
Collecting hvac
  Downloading https://files.pythonhosted.org/packages/e4/70/c7ee5ad9764b23eb9261a8767989ad933abca99c24bf2c3dd6e23e062a3d/hvac-0.11.0-py2.py3-none-any.whl (148kB)
Collecting six>=1.5.0 (from hvac)
  Downloading https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl
Collecting requests>=2.21.0 (from hvac)
  Downloading https://files.pythonhosted.org/packages/92/96/144f70b972a9c0eabbd4391ef93ccd49d0f2747f4f6a2a2738e99e5adc65/requests-2.26.0-py2.py3-none-any.whl (62kB)
Collecting urllib3<1.27,>=1.21.1 (from requests>=2.21.0->hvac)
  Downloading https://files.pythonhosted.org/packages/5f/64/43575537846896abac0b15c3e5ac678d787a4021e906703f1766bfb8ea11/urllib3-1.26.6-py2.py3-none-any.whl (138kB)
Collecting charset-normalizer~=2.0.0; python_version >= "3" (from requests>=2.21.0->hvac)
  Downloading https://files.pythonhosted.org/packages/48/84/aa70b1e0d9d5a76d3d8a4c3d495f8f0524831571f65efe51bb8db8df0eed/charset_normalizer-2.0.5-py3-none-any.whl
Collecting idna<4,>=2.5; python_version >= "3" (from requests>=2.21.0->hvac)
  Downloading https://files.pythonhosted.org/packages/d7/77/ff688d1504cdc4db2a938e2b7b9adee5dd52e34efbd2431051efc9984de9/idna-3.2-py3-none-any.whl (59kB)
Collecting certifi>=2017.4.17 (from requests>=2.21.0->hvac)
  Downloading https://files.pythonhosted.org/packages/05/1b/0a0dece0e8aa492a6ec9e4ad2fe366b511558cdc73fd3abc82ba7348e875/certifi-2021.5.30-py2.py3-none-any.whl (145kB)
Installing collected packages: six, urllib3, charset-normalizer, idna, certifi, requests, hvac
Successfully installed certifi charset-normalizer hvac idna requests-2.9.1 six-1.10.0 urllib3-1.13.1
+ cd asoc
+ cp **** ./AppScanJobs.txt
+ python3 asoc.py
Traceback (most recent call last):
  File "/home/jenkins/.local/lib/python3.5/site-packages/charset_normalizer/api.py", line 5, in <module>
    from os import PathLike
ImportError: cannot import name 'PathLike'
```

**Desktop (please complete the following information):**
 - OS: [Linux]
 - Python version [3.5]
 - Package version [2.0.5]

**Additional context**

Before the 14th
```
+ pip3 install hvac --user --disable-pip-version-check
Collecting hvac
  Downloading https://files.pythonhosted.org/packages/e4/70/c7ee5ad9764b23eb9261a8767989ad933abca99c24bf2c3dd6e23e062a3d/hvac-0.11.0-py2.py3-none-any.whl (148kB)
Collecting requests>=2.21.0 (from hvac)
  Downloading https://files.pythonhosted.org/packages/92/96/144f70b972a9c0eabbd4391ef93ccd49d0f2747f4f6a2a2738e99e5adc65/requests-2.26.0-py2.py3-none-any.whl (62kB)
Collecting six>=1.5.0 (from hvac)
  Downloading https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl
Collecting urllib3<1.27,>=1.21.1 (from requests>=2.21.0->hvac)
  Downloading https://files.pythonhosted.org/packages/5f/64/43575537846896abac0b15c3e5ac678d787a4021e906703f1766bfb8ea11/urllib3-1.26.6-py2.py3-none-any.whl (138kB)
Collecting idna<4,>=2.5; python_version >= "3" (from requests>=2.21.0->hvac)
  Downloading https://files.pythonhosted.org/packages/d7/77/ff688d1504cdc4db2a938e2b7b9adee5dd52e34efbd2431051efc9984de9/idna-3.2-py3-none-any.whl (59kB)
Collecting charset-normalizer~=2.0.0; python_version >= "3" (from requests>=2.21.0->hvac)
  Downloading https://files.pythonhosted.org/packages/33/53/b7f6126a2b9fd878b025fe3c40266cfaad696f312165008ce045bffa3fe7/charset_normalizer-2.0.4-py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests>=2.21.0->hvac)
  Downloading https://files.pythonhosted.org/packages/05/1b/0a0dece0e8aa492a6ec9e4ad2fe366b511558cdc73fd3abc82ba7348e875/certifi-2021.5.30-py2.py3-none-any.whl (145kB)
Installing collected packages: urllib3, idna, charset-normalizer, certifi, requests, six, hvac
Successfully installed certifi charset-normalizer hvac idna requests-2.9.1 six-1.10.0 urllib3-1.13.1
+ cd asoc
+ cp **** ./AppScanJobs.txt
+ python3 asoc.py
... Deprecation Warnings ...
app_id = XXX
scan_id = XXX
execution_id = XXX
Status = Running
Progress = 1
Continue to wait for completion...
Status = Running
Progress = 1
Continue to wait for completion...
Status = Running
Progress = 1
Continue to wait for completion...
Status = Ready
Progress = 100
New issues found = 0
...
```

**Comment by bsiper** [comment]
This is entirely due to the use of [`os.Pathlike`](https://docs.python.org/3/library/os.html#os.PathLike), which does not exist in <3.6.

version 2.0.5 should probably note that it is no longer compatible with 3.5.

**Comment by Ousret** [comment]
Thanks for the report, but you are a bit mistaken.
Yes I accidentally broke the compatibility with some older version but not all 3.5.x

@bsiper you are right, there is a fix available in dev-master and the PR explain everything that we should know about this.
I will publish a tag as soon as possible but I have to run some heavy testing beforehand. 

Meanwhile, I suggest you to either pin `charset_normalizer` to 2.0.4 or upgrade your Python to latest minor 3.5.10.

**Comment by Ousret** [comment]
Now fixed and live v2.0.6
